holoviews/element/graphs.py

```diff
@@ -0,0 +1,259 @@
+            if self._edgepaths:
+                path_range = self._edgepaths.range(dimension, data_range)
+                return max_range([node_range, path_range])
+            return node_range
+        return super(Graph, self).range(dimension, data_range)
+
+
+    def dimensions(self, selection='all', label=False):
+        dimensions = super(Graph, self).dimensions(selection, label)
+        if selection == 'ranges':
+            if self._nodes:
+                node_dims = self.nodes.dimensions(selection, label)
+            else:
+                node_dims = Nodes.kdims+Nodes.vdims
+                if label in ['name', True, 'short']:
+                    node_dims = [d.name for d in node_dims]
+                elif label in ['long', 'label']:
+                    node_dims = [d.label for d in node_dims]
+            return dimensions+node_dims
+        return dimensions
+
+
+    @property
+    def nodes(self):
+        """
+        Computes the node positions the first time they are requested
+        if no explicit node information was supplied.
+        """
+        if self._nodes is None:
+            self._nodes = layout_nodes(self)
+        return self._nodes
+
+
+    @property
+    def edgepaths(self):
+        """
+        Returns the fixed EdgePaths or computes direct connections
+        between supplied nodes.
+        """
+        if self._edgepaths:
+            return self._edgepaths
+        paths = []
+        for start, end in self.array(self.kdims):
+            start_ds = self.nodes[:, :, start]
+            end_ds = self.nodes[:, :, end]
+            sx, sy = start_ds.array(start_ds.kdims[:2]).T
+            ex, ey = end_ds.array(end_ds.kdims[:2]).T
+            paths.append([(sx[0], sy[0]), (ex[0], ey[0])])
+        return EdgePaths(paths, kdims=self.nodes.kdims[:2])
+
+
+    @classmethod
+    def from_networkx(cls, G, layout_function, nodes=None, **kwargs):
+        """
+        Generate a HoloViews Graph from a networkx.Graph object and
+        networkx layout function. Any keyword arguments will be passed
+        to the layout function.
+        """
+        positions = layout_function(G, **kwargs)
+        if nodes:
+            xs, ys = zip(*[v for k, v in sorted(positions.items())])
+            nodes = nodes.add_dimension('x', 0, xs)
+            nodes = nodes.add_dimension('y', 1, ys).clone(new_type=Nodes)
+        else:
+            nodes = Nodes([tuple(pos)+(idx,) for idx, pos in sorted(positions.items())])
+        return cls((G.edges(), nodes))
+
+
+class Nodes(Points):
+    """
+    Nodes is a simple Element representing Graph nodes as a set of
+    Points.  Unlike regular Points, Nodes must define a third key
+    dimension corresponding to the node index.
+    """
+
+    kdims = param.List(default=[Dimension('x'), Dimension('y'),
+                                Dimension('index')], bounds=(3, 3))
+
+    group = param.String(default='Nodes', constant=True)
+
+
+class EdgePaths(Path):
```

Comment:
I'm wondering whether it is worth introducing ``EdgePaths``. Why not just use ``Path``?